serpy2/serpy2.py

```diff
@@ -0,0 +1,552 @@
+            # at least STIMEOUT
+            if not self.out_q.empty():
+                bdata, btype = self.out_q.get()
+                # Following conserved from serpy original and commentized
+                # illustrating handshaking via events
+                #
+                # if mode != self.out_mode :
+                #     self.modeChangeEvent.clear()
+                #     self.out_block_q.put(b"\x01SETMODE" +
+                #                         str(mode).encode(self.encoding))
+                #     self.modeChangeEvent.wait()
+                #     self.out_mode = mode
+                
+                # non-blocking (the only implemented mode)
+                #TODO introduce blocking send
+                # should best be implemented probably 
+                # in the SendData via a join thread or
+                # wait for an empty queue event?
+                
+                # encode header
+                checksum = sum(bdata) & 0xFFFFFFFF
+                datacheckb = checksum.to_bytes(4, 'little', signed = False)
+                datalenb = len(bdata).to_bytes(4, 'little', signed = False)
+
+                bhead = BLOCKIDCODE + btype + datacheckb + datalenb
+                assert len(bhead) == HEADLEN, 'HEADLEN incorrect'
+                block = bhead + bdata
+                
+                self.out_block_q.put(block)
+            else:
+                sleep(STIMEOUT)
+                
+    
+    def getData(self, timeout=None):
+        """
+        Returns the received data in the original data type
+        Optional argument : timeout (default None)
+        
+        Will block until there is data or timeout is reached.
+        On timeout will raise queue.Empty exception
+        """
+        return self.in_q.get(timeout=timeout)
+    
+    def sendData(self, data, timeout=None):
+        """
+        Send the data over this connection following a mode.
+        
+        Optional timeout (in seconds [float]) may be specified. 
+        May block for a maximum of 2*timeout.
+        May raise raise queue.Full exception or return False on timeout
+        ------
+        
+        Currently supports the following types of data
+        
+        str  : sent as utf-8 encoded bytes
+        int  : sent as signed 64-bit integer encoded with int.to_bytes()
+        float: sent as ascii-encoded bytes after representation with float.hex()
+        bytes: sent as raw data
+        """
+        if type(data) == float:
+            flhex = data.hex()
+            bdata = flhex.encode('ascii')
+            btype = FLOATTYPECODE
+        elif type(data) == int:
+            bdata = data.to_bytes(8, 'little', signed = True)
+            btype = INTTYPECODE
+        elif type(data) == str:
+            bdata = data.encode('utf-8')
+            btype = STRTYPECODE
+        elif type(data) == bytes:
+            bdata = data
+            btype = RAWBYTESTYPECODE
```

Comment:
Probably should use `isinstance` in case data is a subclass

Reply:
Thanks for pointing this out (see also: https://switowski.com/blog/type-vs-isinstance).
However, for now, let's work only with the base classes `float`, `int`, `bytes`, `str` and no sub-classes.
I will change the `==` in `is`.